File: codeforces/72.honest_coach_[1360B]/honest_coach.cpp
#include <iostream>
#include <algorithm>
#include <vector>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned short int t, n, si, pos;
  vector<short int> s;
  vector<short int> r;
  cin >> t;
  while (t--) {
    cin >> n;
    for (unsigned short int i = 0; i < n; i++) {
      cin >> si;
      s.push_back(si);
    }
    sort(s.begin(), s.end());
    for (unsigned short int i = 1; i < n; i++) {
      r.push_back(s.at(i) - s.at(i - 1));
    }
    cout << *min_element(r.begin(), r.end()) << endl;
    s.clear();
    r.clear();
  }
  return 0;
}

File: codeforces/low_difficulty/800_points/72.honest_coach_[1360B]/honest_coach.py
# Author: José Rodolfo (jric2002)
t = int(input())
while (t):
  n = int(input())
  s = [int(x) for x in input().split(" ")]
  r = []
  s.sort()
  for i in range(1, n):
    r.append(s[i] - s[i - 1])
  print(min(r))
  t -= 1